scene/register_scene_types.cpp

```diff
@@ -1027,6 +1027,26 @@ void register_scene_types() {
 	GDREGISTER_CLASS(NavigationObstacle2D);
 	GDREGISTER_CLASS(NavigationLink2D);
 
+#ifdef MODULE_NAVIGATION_ENABLED
+	OS::get_singleton()->yield(); // May take time to init.
+
+	// 2D nodes that support navmesh baking need to server register their source geometry parsers.
+	MeshInstance2D::_navmesh_parse_init();
+	MultiMeshInstance2D::_navmesh_parse_init();
+	NavigationObstacle2D::_navmesh_parse_init();
+	Polygon2D::_navmesh_parse_init();
+	TileMap::_navmesh_parse_init();
+	TileMapLayer::_navmesh_parse_init();
+	StaticBody2D::_navmesh_parse_init();
+#ifndef _3D_DISABLED
+	// 3D nodes that support navmesh baking need to server register their source geometry parsers.
+	MeshInstance3D::_navmesh_parse_init();
+	MultiMeshInstance3D::_navmesh_parse_init();
+	NavigationObstacle3D::_navmesh_parse_init();
+	StaticBody3D::_navmesh_parse_init();
+#endif
+#endif
```

Comment:
Can't this be done in `modules/navigation/register_types.cpp`, to avoid breaking encapsulation?

Otherwise this requires including `modules_enabled.gen.h` to actually know that the navigation module is enabled.

Reply:
Well that is a good question. The parsers are actually not part of the navigation module. They are part of the core NavigationServer, not the GodotNavigationServer module part.

Moving to the module would have the side-effect that if the navigation module is theoretically disabled or replaced that all the core node parsers would not register. While it is not as common as with the PhysicsServer (Jolt) that the NavigationServer gets replaced every new NavigationServer override module would need to register all the core node parsers manually again.

The idea behind having the parsers register here was that any kind of navigation module replace does not need to care about that list of nodes as the actual internal parse data used by the server is always the same.

Reply:
But then it shouldn't be excluded if `MODULE_NAVIGATION_ENABLED` isn't defined, no? This is specifically checking for the GodotNavigationServer module.

Reply:
Indeed there is some flawed logic here.